cypress/integration/SourceTest.ts

```diff
@@ -0,0 +1,214 @@
+        .find("option")
+        .should("have.length", source.length)
+        .each((item, index) => {
+          cy.wrap(item).should("have.text", source[index]);
+        });
+    });
+
+    it("The default state of the configuration", () => {
+      cy.ouiaId("missing-sources", "PF4/TextInput")
+        .scrollIntoView()
+        .should("be.visible")
+        .should("be.disabled");
+      cy.ouiaId("configuration").should("not.exist");
+    });
+
+    describe("Slack Source", () => {
+      beforeEach(() => {
+        cy.ouiaId("source-type", "PF4/FormSelect")
+          .should("be.visible")
+          .select("Slack Source");
+        cy.ouiaId("missing-sources", "PF4/TextInput").should("not.exist");
+        cy.ouiaId("configuration").should("be.visible");
+      });
+
+      it("Check parameters", () => {
+        cy.ouiaId("configuration").within(() => {
+          cy.ouiaType("config-parameter").should("have.length", 4);
+          cy.ouiaId("slack_channel", "config-parameter")
+            .should("be.visible")
+            .should("contain", "Example: #myroom");
+          cy.ouiaId("slack_token", "config-parameter").should("be.visible");
+          cy.ouiaId("slack_delay", "config-parameter")
+            .should("be.visible")
+            .should("contain", "Example: 1s");
+          cy.ouiaId("kafka_topic", "config-parameter").should("be.visible");
+        });
+      });
+
+      it("Save the required parameter without any value", () => {
+        cy.ouiaId("submit").click();
+        cy.ouiaId("configuration")
+          .should("be.visible")
+          .scrollIntoView()
+          .within(() => {
+            //required parameters
+            cy.ouiaId("slack_channel", "config-parameter").within(() => {
+              cy.get("div.pf-m-error")
+                .should("be.visible")
+                .should(
+                  "contain",
+                  "must have required property 'slack_channel'"
+                );
+              //insert any value
+              cy.root().type("#aaa");
+
+              cy.get("div.pf-m-error").should("not.exist");
+            });
+            cy.ouiaId("slack_token", "config-parameter").within(() => {
+              cy.get("div.pf-m-error")
+                .should("be.visible")
+                .should("contain", "must have required property 'slack_token'");
+              //insert any value
+              cy.root().type("ccc");
+              cy.get("div.pf-m-error").should("not.exist");
+            });
+
+            //optional parameters
+            cy.ouiaId("slack_delay", "config-parameter")
+              .should("be.visible")
+              .find("div.pf-m-error")
+              .should("not.exist");
+            cy.ouiaId("kafka_topic", "config-parameter")
+              .should("be.visible")
+              .find("div.pf-m-error")
+              .should("not.exist");
+          });
+      });
+    });
+
+    describe("Dummy Source", () => {
```

Comment:
Hi @sjamboro, could you please clarify why you introduced a new dummy source schema? Thanks

Reply:
@kelvah the reason is that all sources contains lots of parameters and all of them contain token as required. The action list contains "Send to bridge" with has only one optional parameter. The dummy source is ideal for mock tests to check if number of parameters was changed.

Reply:
It looks a bit overkill to me, but it seems it makes life easier for testing. Approving then. Thank you.